File: src/gestorAplicacion/tienda/CajaRegistradora.java
package gestorAplicacion.tienda;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import gestorAplicacion.personal.*;
/**
 * 
 * @author Erik Gonzalez
 * @summary En la caja registrado se lleva la contabilidad de los servicios, con el total de ingresos y
 * los servicios prestados
*/
public class CajaRegistradora implements Serializable {
	private static final long serialVersionUID = 1L;
	private double totalIngresos;
	private List<Servicio> servicios;
	
	public static List<CajaRegistradora> cajasRegistradoras;
	
	static {
		cajasRegistradoras = new ArrayList<CajaRegistradora>();
	}
	
	public CajaRegistradora() {
		servicios = new ArrayList<Servicio>();
		cajasRegistradoras.add(this);
	}
	
	/**
	 * Registra la venta cuando finaliza el servico
	 * @param precio
	 * @param servicio
	 */
	public void registrarVenta(double precio, Servicio servicio) {
		servicios.add(servicio);
		this.totalIngresos += precio;
	}
	
	public double getTotalIngresos() {
		return this.totalIngresos;
	}

	public void setTotalIngresos(double totalIngresos) {
		this.totalIngresos = totalIngresos;
	}
	
	public double descontar(double porcentaje) {
		double descuento = this.totalIngresos*porcentaje;
		this.totalIngresos-= descuento;
		return descuento;
	}
}

File: practica2/generic_it/gestor_aplicacion/tienda/caja_registradora.py
'''
@File    :   caja_registradora.py
@Time    :   2022/01/26
@Author  :   Erik Gonzalez
@Version :   1.0
@Desc    :   La caja registradora lleva la contabilidad de la empresa
'''


class CajaRegistradora:
    cajasRegistradoras = []

    def __init__(self):
        self._servicios = []
        self._totalIngresos = 0
        CajaRegistradora.cajasRegistradoras.append(self)

    def registrarVenta(self, precio, servicio):
        self._servicios.append(servicio)
        self._totalIngresos += precio

    def getTotalIngresos(self):
        return self._totalIngresos

    def setTotalIngresos(self, totalIngresos):
        self._totalIngresos = totalIngresos

    def descontar(self, porcentaje):
        return self._totalIngresos * porcentaje
